guidedmodules/views.py

```diff
@@ -426,6 +428,113 @@ def redirect_to():
+                # Only two actions are currently supported:
+                #   1. `element/add_role/<role_value>` - Automatically add elements to the selected components of a system
+                #   2. `element/del_role/<role_value>` - Automatically delete elements from the selected components of a system
+                if a_obj == 'element':
+
+                    # Get all elements assigned role specified in the action
+                    elements_with_role = Element.objects.filter(roles__role=a_filter)
+
+                    # Add elements matching role to the selected components of a system
+                    if a_verb == "add_role":
+                        # TODO: Optimize and improve DRY-ness of this code block
+                        # The adding of a component code is copied from `controls.views.add_system_component`.
+                        # It was not possible to easily combine the code because
+                        # the request and http POST information `add_system_component` uses.
+                        # Also, `add_system_component` has different assumptions about routing.
+                        #
+                        # Nevertheless, the two code blocks could be refactored.
+
+                        # Get system's existing components selected because we don't want to add an already added component
+                        elements_selected = system.producer_elements
+                        elements_selected_ids = [e.id for e in elements_selected]
+
+                        # Get system's selected controls because we only want to add statements for selected controls
+                        # TODO DO WE ONLY WANT TO ADD STATEMENTS FOR EXISTING CONTROLS?
+                        selected_controls = system.root_element.controls.all()
+                        selected_controls_ids = set([f"{sc.oscal_ctl_id} {sc.oscal_catalog_key}" for sc in selected_controls])
+                        # TODO: Refactor above line selected_controls into a system model function if not already existing
+
+                        # Iterate through elements in role adding each to the selected components of the system
+                        for producer_element in elements_with_role:
+                            # TODO: various use cases
+                                # - component previously added but element has statements not yet added to system
+                                #   this issue may be best addressed elsewhere.
+                            # Component already added to system. Do not add the component (element) to the system again.
+                            if producer_element.id in elements_selected_ids:
+                                messages.add_message(request, messages.ERROR,
+                                                    f'Component "{producer_element.name}" already exists in selected components.')
+                                # Go to next element
+                                next
```

Comment:
```suggestion
                                continue
```
next by itself doesn't do anything I think you wanted to use continue. Same for below.